### PR TITLE
routing-daemon/README.md: Various fixes

### PR DESCRIPTION
**Pull request by Miciah**
Fix a grammatical error in the "Configuring the Daemon" section.

Delete `BIGIP_MONITOR` from the list of settings for the F5 BIG-IP LTM controller, and add `BIGIP_SSHKEY` and `BIGIP_DEVICE_GROUP`.

Make it clear that the HTTP virtual server and the HTTPS virtual server may share a vip.

Reorganise the documentation for F5 BIG-IP LTM for readability.

Document the `BIGIP_DEVICE_GROUP` setting.

Correct the description of the local-traffic policy is created to say that the daemon creates it on initialisation and adds and removes rules during operation, and mention application aliases and SSL certificates and keys as well.

Add `NGINX_SSL_CERTIFICATE`, `NGINX_SSL_KEY`, `HTTP_PORT`, and `SSL_PORT` to the list of settings for the nginx controller.

Mention the nginx configuration files for application aliases and SSL configuration.

Document the `HTTP_PORT` and `SSL_PORT` settings.

### CODE REVIEW

File: routing-daemon/README.md [hunk 1]
```diff
@@ -44,27 +44,46 @@ Using F5 BIG-IP LTM
...
-in `/etc/openshift/routing-daemon.conf` to match the names you've used. The
-daemon will name the pools after applications following the template
-"/Common/ose-#{app_name}-#{namespace}" and create policy rules to forward
-requests to pools comprising the gears of the named application.
+and one for HTTPS traffic. Each virtual server needs to be assigned at least one
+VIP.  It is not necessary that each virtual server have a unique VIP: the HTTP
+virtual server and the HTTPS virtual servers may share a VIP.  Once the LTM
+virtual servers have been created, update `VIRTUAL_SERVER` and
+`VIRTUAL_HTTPS_SERVER` in `/etc/openshift/routing-daemon.conf` to match the
+names you've used.
```
Comment (ahardin-rh) [inline routing-daemon/README.md]: s/you've/you have

File: routing-daemon/README.md [hunk 2]
```diff
@@ -44,27 +44,46 @@ Using F5 BIG-IP LTM
...
+
+A default client-ssl profile must also be configured as the default SNI
+client-ssl profile. Although the naming of the default client-ssl profile is
+unimportant, it does need to be added to the HTTPS virtual server.
+
+The LTM admin user's 'Terminal Access' must be set to 'Advanced shell' so that
+remote shell commands may be executed. Additionally, for the remote key
+management commands to execute, the `BIGIP_SSHKEY` public key must be added to
+the LTM admin's `.ssh/authorized_keys` file.
```
Comment (ahardin-rh) [inline routing-daemon/README.md]: s/admin/administrator  (2 instances)

File: routing-daemon/README.md [hunk 3]
```diff
@@ -44,27 +44,46 @@ Using F5 BIG-IP LTM
...
+virtual server and the HTTPS virtual servers may share a VIP.  Once the LTM
+virtual servers have been created, update `VIRTUAL_SERVER` and
+`VIRTUAL_HTTPS_SERVER` in `/etc/openshift/routing-daemon.conf` to match the
+names you've used.
+
+A default client-ssl profile must also be configured as the default SNI
+client-ssl profile. Although the naming of the default client-ssl profile is
+unimportant, it does need to be added to the HTTPS virtual server.
+
+The LTM admin user's 'Terminal Access' must be set to 'Advanced shell' so that
```
Comment (ahardin-rh) [inline routing-daemon/README.md]: Should  "shell" also be capitalized?

File: routing-daemon/README.md [hunk 4]
```diff
@@ -112,8 +138,8 @@ Set `MONITOR_UP_CODE` to the code that indicates that a pool member is up, or
 leave `MONITOR_UP_CODE` unset to use the default value of "1".
 
 Set `MONITOR_TYPE` to either "http-ecv" or "https-ecv" depending on whether you
```
Comment (ahardin-rh) [inline routing-daemon/README.md]: add a comma before the word "depending"

File: routing-daemon/README.md [hunk 5]
```diff
@@ -134,6 +160,25 @@ for the associated monitor, and that the host will respond with the value of
 `MONITOR_UP_CODE` if the host is up or some other response if the host is not
```
Comment (ahardin-rh) [inline routing-daemon/README.md]: add a comma between up and or

File: routing-daemon/README.md [hunk 6]
```diff
@@ -134,6 +160,25 @@ for the associated monitor, and that the host will respond with the value of
...
 up.
 
+Endpoint Types
+--------------
+
+By default, the routing daemon adds only proxy gears to pools.  Specifically,
+the routing daemon ignores any gear endpoints that do not have the
+"load_balancer" type.  These "load_balancer" endpoints are the endpoints for
+HAProxy gears. Thus requests coming into the external load-balancer (nginx or F5
```
Comment (ahardin-rh) [inline routing-daemon/README.md]: Add a comma after Thus

File: routing-daemon/README.md [hunk 7]
```diff
@@ -134,6 +160,25 @@ for the associated monitor, and that the host will respond with the value of
...
+BIG-IP) will be routed through applications' HAProxy gears to reach the
+application gears.
+
+Routing requests through HAProxy gears enables the auto-scaling logic to react
+to changes in load.  However, because only scalable applications have HAProxy
+gears, this approach also means that only scalable applications can be reached
+through the external load-balancer.
+
+The `ENDPOINT_TYPES` setting specifies which gear/endpoint types the routing
```
Comment (ahardin-rh) [inline routing-daemon/README.md]: gear or endpoint instead of gear/endpoint

**Comment by ahardin-rh** [comment]
@Miciah Just a few nits inline. Other than those little things, LGTM.

File: routing-daemon/README.md (hunk 2 above)
Comment (Miciah) [inline routing-daemon/README.md]: "admin" refers to the name of the administrative user.  I will put it in quotation marks to make that clearer.

File: routing-daemon/README.md (hunk 3 above)
Comment (Miciah) [inline routing-daemon/README.md]: "Terminal Access" and "Advanced shell" come directly from the management console, so I am preserving the capitalisation used there: http://network-cisco.up.n.seesaa.net/network-cisco/image/WS000033-5ec7d.JPG

**Comment by Miciah** [comment]
Thanks! Updated per your comments.

**Comment by Miciah** [comment]
openshift-bot, please [merge]!

**Pull request merged**